test/tokenVesting.ts

```diff
@@ -0,0 +1,319 @@
+    let block = await ethers.provider.getBlock("latest");
+    let blockTime = block.timestamp;
+    let startTime = new Date(blockTime * 1000);
+    let endTime = new Date(blockTime * 1000);
+
+    // start after 3 months
+    startTime.setMonth(startTime.getMonth() + 3);
+    // end after 3 months of start
+    endTime.setMonth(startTime.getMonth() + 3);
+
+    const startTimeStamp = getUnixTimeStamp(startTime);
+    const endTimeStamp = getUnixTimeStamp(endTime);
+
+    const account1 = await accounts[3].getAddress();
+    const account2 = await accounts[4].getAddress();
+    const dummyCommunity = [account1, account2];
+    const dummyAmounts = [amount1, amount2];
+    const immidiateRelease = BigNumber.from("0");
+
+    await TokenVesting.connect(accounts[0]).addVestings(
+      dummyCommunity,
+      dummyAmounts,
+      startTimeStamp.toString(),
+      endTimeStamp.toString(),
+      immidiateRelease,
+    );
+
+    expect(await TokenVesting.claimable(dummyCommunity[0])).to.be.equal(0);
+    expect(await TokenVesting.claimable(dummyCommunity[1])).to.be.equal(0);
+
+    await increaseBlockTime(avg_seconds_in_months);
+    await mineOneBlock();
+
+    expect(await TokenVesting.claimable(dummyCommunity[0])).to.be.equal(0);
+    expect(await TokenVesting.claimable(dummyCommunity[1])).to.be.equal(0);
+
+    await increaseBlockTime(avg_seconds_in_months);
+    await mineOneBlock();
+
+    expect(await TokenVesting.claimable(dummyCommunity[0])).to.be.equal(0);
+    expect(await TokenVesting.claimable(dummyCommunity[1])).to.be.equal(0);
+
+    await increaseBlockTime(avg_seconds_in_months);
+    await mineOneBlock();
+
+    expect(Math.floor((await TokenVesting.claimable(dummyCommunity[0])).toNumber() / 1000)).to.be.equal(0);
+    expect(Math.floor((await TokenVesting.claimable(dummyCommunity[1])).toNumber() / 1000)).to.be.equal(0);
+
+    await increaseBlockTime(avg_seconds_in_months);
+    await mineOneBlock();
+
+    expect(Math.floor((await TokenVesting.claimable(dummyCommunity[0])).toNumber() / 1000)).to.be.equal(3);
+    expect(Math.floor((await TokenVesting.claimable(dummyCommunity[1])).toNumber() / 1000)).to.be.equal(3);
+  });
+
+  it("should vest with immidiate Release", async function () {
+    const amount1 = await ethers.utils.parseEther("0.00000000000001");
+    const amount2 = await ethers.utils.parseEther("0.00000000000001");
+    let block = await ethers.provider.getBlock("latest");
+    let blockTime = block.timestamp;
+    let startTime = new Date(blockTime * 1000);
+    let endTime = new Date(blockTime * 1000);
+
+    endTime.setMonth(startTime.getMonth() + 3);
+
+    const startTimeStamp = getUnixTimeStamp(startTime);
+    const endTimeStamp = getUnixTimeStamp(endTime);
+
+    const account1 = await accounts[3].getAddress();
+    const account2 = await accounts[4].getAddress();
+    const dummyCommunity = [account1, account2];
+    const dummyAmounts = [amount1, amount2];
+    const immidiateRelease = BigNumber.from("10");
```

Comment:
maybe we should declare immediateRelease as a global variable and mutate it within testcases